listen.go

```diff
@@ -0,0 +1,100 @@
+package tlsdefaults
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/getlantern/keyman"
+)
+
+var (
+	tenYearsFromToday = time.Now().AddDate(10, 0, 0)
+)
+
+// Listen opens a TLS listener at the given address using the private key and
+// certificate PEM files at the given paths. If no files exists, it creates a
+// new key and self-signed certificate at those locations.
+func Listen(addr, pkfile, certfile string) (net.Listener, error) {
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to split host and port for %v: %v\n", addr, err)
+	}
+
+	mypkfile := pkfile
+	if mypkfile == "" {
+		mypkfile = "key.pem"
```

Comment:
Can we just return error if no file name provided? Caller will pass these two parameters anyway.